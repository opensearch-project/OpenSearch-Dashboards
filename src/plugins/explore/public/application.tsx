/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import ReactDOM from 'react-dom';
import { AppMountParameters, CoreStart } from '../../../core/public';
import { ExploreStartPlugins } from './types';
import { ExploreApp } from './components/app';

export const renderApp = (
  { notifications, http }: CoreStart,
  { navigation }: ExploreStartPlugins,
  { appBasePath, element }: AppMountParameters
) => {
  ReactDOM.render(
    <ExploreApp
      basename={appBasePath}
      notifications={notifications}
      http={http}
      navigation={navigation}
    />,
    element
  );

  return () => ReactDOM.unmountComponentAtNode(element);
};
