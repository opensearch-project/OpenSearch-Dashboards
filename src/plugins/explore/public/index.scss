/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

// Import the new layout styles
@import "./application/components/explore_layout";
