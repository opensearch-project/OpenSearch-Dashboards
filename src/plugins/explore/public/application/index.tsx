/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import ReactDOM from 'react-dom';
import { Router, Route, Switch } from 'react-router-dom';
import { Provider as ReduxProvider } from 'react-redux';
import { Store } from 'redux';
import { AppMountParameters } from '../../../../core/public';
import { ExploreServices } from '../types';
import { ExploreApp } from './app';
import { OpenSearchDashboardsContextProvider } from '../../../opensearch_dashboards_react/public';
import { IndexPatternProvider } from './components/index_pattern_context';

// Route component props interface
interface ExploreRouteProps {
  services: ExploreServices;
  history: AppMountParameters['history'];
}

// Route components for different paths
const ExploreMainRoute = (props: ExploreRouteProps & { setHeaderActionMenu?: any }) => (
  <ExploreApp setHeaderActionMenu={props.setHeaderActionMenu} />
);

// View route for saved searches
const ViewRoute = (props: ExploreRouteProps & { setHeaderActionMenu?: any }) => (
  <ExploreApp setHeaderActionMenu={props.setHeaderActionMenu} />
);

export const renderApp = (
  { element, history, setHeaderActionMenu }: AppMountParameters,
  services: ExploreServices,
  store: Store
) => {
  // Create main route props
  const mainRouteProps = {
    services,
    history,
    setHeaderActionMenu,
  };
  ReactDOM.render(
    <Router history={history}>
      <OpenSearchDashboardsContextProvider services={services}>
        <ReduxProvider store={store}>
          <IndexPatternProvider>
            <services.core.i18n.Context>
              <Switch>
                {/* View route for saved searches */}
                <Route path="/view/:id" exact>
                  <ViewRoute {...mainRouteProps} />
                </Route>

                {/* Main route */}
                <Route path="/" exact>
                  <ExploreMainRoute {...mainRouteProps} />
                </Route>

                {/* Default fallback */}
                <Route>
                  <ExploreMainRoute {...mainRouteProps} />
                </Route>
              </Switch>
            </services.core.i18n.Context>
          </IndexPatternProvider>
        </ReduxProvider>
      </OpenSearchDashboardsContextProvider>
    </Router>,
    element
  );

  return () => {
    ReactDOM.unmountComponentAtNode(element);
  };
};
