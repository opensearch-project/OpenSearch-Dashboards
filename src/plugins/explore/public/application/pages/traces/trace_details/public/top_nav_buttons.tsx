/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React, { useState } from 'react';
import { i18n } from '@osd/i18n';
import {
  EuiOverlayMask,
  EuiModal,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiModalBody,
  EuiModalFooter,
  EuiButton,
  EuiCodeBlock,
} from '@elastic/eui';
import type { MountPoint } from 'opensearch-dashboards/public';
import { TopNavMenu } from '../../../../../../../navigation/public';
import type { TopNavMenuData } from '../../../../../../../navigation/public';

export interface TraceTopNavMenuProps {
  payloadData: any[];
  setMenuMountPoint?: (mount: MountPoint | undefined) => void;
}

export const TraceTopNavMenu: React.FC<TraceTopNavMenuProps> = ({
  payloadData,
  setMenuMountPoint,
}) => {
  const [isRawModalOpen, setRawModalOpen] = useState(false);

  const menuActions: TopNavMenuData[] = [
    {
      id: 'viewRawData',
      label: i18n.translate('explore.traceDetails.topNav.viewRawData', {
        defaultMessage: 'View raw trace',
      }),
      run: () => setRawModalOpen(true),
      testId: 'viewRawDataBtn',
      emphasize: true,
    },
  ];

  return (
    <>
      <TopNavMenu
        config={menuActions}
        setMenuMountPoint={setMenuMountPoint}
        showSearchBar={false}
        showQueryBar={false}
        showQueryInput={false}
        showDatePicker={false}
        showFilterBar={false}
        showDataSourceMenu={false}
        appName={'TraceDetails'}
      />
      {isRawModalOpen && (
        <EuiOverlayMask>
          <EuiModal onClose={() => setRawModalOpen(false)}>
            <EuiModalHeader>
              <EuiModalHeaderTitle>
                {i18n.translate('explore.traceDetails.modal.rawDataTitle', {
                  defaultMessage: 'Raw data',
                })}
              </EuiModalHeaderTitle>
            </EuiModalHeader>
            <EuiModalBody>
              {payloadData && payloadData.length > 0 && (
                <EuiCodeBlock language="json" paddingSize="s" isCopyable overflowHeight={500}>
                  {JSON.stringify(payloadData, null, 2)}
                </EuiCodeBlock>
              )}
            </EuiModalBody>
            <EuiModalFooter>
              <EuiButton onClick={() => setRawModalOpen(false)} fill>
                {i18n.translate('explore.traceDetails.modal.closeButton', {
                  defaultMessage: 'Close',
                })}
              </EuiButton>
            </EuiModalFooter>
          </EuiModal>
        </EuiOverlayMask>
      )}
    </>
  );
};
