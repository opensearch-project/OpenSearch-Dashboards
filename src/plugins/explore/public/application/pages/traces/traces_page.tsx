/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import './traces_page.scss';

import React, { useRef } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  EuiErrorBoundary,
  EuiFlexGroup,
  EuiFlexItem,
  EuiResizableContainer,
  EuiPage,
  EuiPageBody,
  EuiSpacer,
  useIsWithinBreakpoints,
} from '@elastic/eui';
import { AppMountParameters, HeaderVariant } from 'opensearch-dashboards/public';
import { useOpenSearchDashboards } from '../../../../../opensearch_dashboards_react/public';
import { ExploreServices } from '../../../types';
import { RootState } from '../../utils/state_management/store';
import { ResultStatus } from '../../utils/state_management/types';
import {
  TopNav,
  TopNavProps,
} from '../../legacy/discover/application/view_components/canvas/top_nav';
import { DiscoverChartContainer } from '../../legacy/discover/application/view_components/canvas/discover_chart_container';
import { QueryPanel } from '../../../components/query_panel/query_panel';
import { DiscoverPanel } from '../../legacy/discover/application/view_components/panel';
import { HeaderDatasetSelector } from '../../components/header_dataset_selector';
import { useInitialQueryExecution } from '../../utils/hooks/use_initial_query_execution';
import { useUrlStateSync } from '../../utils/hooks/use_url_state_sync';
import { useTimefilterSubscription } from '../../utils/hooks/use_timefilter_subscription';
import { ExploreTabs } from '../../../components/tabs/tabs';
import { useHeaderVariants } from '../../utils/hooks/use_header_variants';
import { NewExperienceBanner } from '../../../components/experience_banners/new_experience_banner';
import { useIndexPatternContext } from '../../components/index_pattern_context';
import { DiscoverNoIndexPatterns } from '../../legacy/discover/application/components/no_index_patterns/no_index_patterns';
import { DiscoverUninitialized } from '../../legacy/discover/application/components/uninitialized/uninitialized';
import { LoadingSpinner } from '../../legacy/discover/application/components/loading_spinner/loading_spinner';
import { DiscoverNoResults } from '../../legacy/discover/application/components/no_results/no_results';
import {
  executeQueries,
  defaultPrepareQuery,
} from '../../utils/state_management/actions/query_actions';
import { CanvasPanel } from '../../legacy/discover/application/components/panel/canvas_panel';
import { selectShowDataSetFields } from '../../utils/state_management/selectors';

/**
 * Main application component for the Explore plugin
 */
export const TracesPage: React.FC<Partial<Pick<AppMountParameters, 'setHeaderActionMenu'>>> = ({
  setHeaderActionMenu,
}) => {
  const { services } = useOpenSearchDashboards<ExploreServices>();
  const dispatch = useDispatch();
  const {
    indexPattern,
    isLoading: indexPatternLoading,
    error: indexPatternError,
  } = useIndexPatternContext();
  const showActionsInGroup = services.uiSettings.get('home:useNewHomePage', false);

  // Get status for conditional rendering
  const status = useSelector((state: RootState) => {
    return state.ui?.status || ResultStatus.UNINITIALIZED;
  });
  const rows = useSelector((state: RootState) => {
    const query = state.query;
    const results = state.results;

    // Use default cache key computation - pass query string only
    const queryString = typeof query.query === 'string' ? query.query : '';
    const cacheKey = defaultPrepareQuery(queryString);
    const rawResults = cacheKey ? results[cacheKey] : null;

    if (rawResults) {
      const hits = rawResults.hits?.hits || [];
      return hits;
    }

    return [];
  });

  const showDataSetFields = useSelector(selectShowDataSetFields);

  const isMobile = useIsWithinBreakpoints(['xs', 's', 'm']);

  useInitialQueryExecution(services);
  useUrlStateSync(services);
  useTimefilterSubscription(services);
  useHeaderVariants(services, HeaderVariant.APPLICATION);

  // TODO: Clean out refs for portal positioning if not needed.
  const topLinkRef = useRef<HTMLDivElement>(null);
  const datasetSelectorRef = useRef<HTMLDivElement>(null);
  const datePickerRef = useRef<HTMLDivElement>(null);

  // Create TopNav props - use portal approach for precise positioning
  const topNavProps: TopNavProps = {
    opts: {
      setHeaderActionMenu: setHeaderActionMenu || (() => {}), // Use real global header mount point
      onQuerySubmit: ({ dateRange, query }: any) => {
        // Update time range
        if (dateRange && services?.data?.query?.timefilter?.timefilter) {
          services.data.query.timefilter.timefilter.setTime(dateRange);
        }
      },
      optionalRef: {
        topLinkRef,
        datasetSelectorRef,
        datePickerRef,
      },
    },
    showSaveQuery: true,
  };

  // Function to refresh data (for uninitialized state)
  const onRefresh = () => {
    if (services) {
      dispatch(executeQueries({ services }) as any);
    }
  };

  const renderBottomRightPanel = () => {
    if (indexPattern == null) {
      return (
        <CanvasPanel>
          <>
            <EuiSpacer size="xxl" />
            <DiscoverNoIndexPatterns />
          </>
        </CanvasPanel>
      );
    }

    if (status === ResultStatus.NO_RESULTS) {
      return (
        <CanvasPanel>
          <DiscoverNoResults
            queryString={services?.data?.query?.queryString}
            query={services?.data?.query?.queryString?.getQuery()}
            savedQuery={services?.data?.query?.savedQueries}
            timeFieldName={indexPattern.timeFieldName}
          />
        </CanvasPanel>
      );
    }

    if (status === ResultStatus.UNINITIALIZED) {
      return (
        <CanvasPanel>
          <DiscoverUninitialized onRefresh={onRefresh} />
        </CanvasPanel>
      );
    }

    if (status === ResultStatus.LOADING && !rows?.length) {
      return (
        <CanvasPanel>
          <LoadingSpinner />
        </CanvasPanel>
      );
    }

    if (status === ResultStatus.ERROR && !rows?.length) {
      return (
        <CanvasPanel>
          <DiscoverUninitialized onRefresh={onRefresh} />
        </CanvasPanel>
      );
    }

    if (
      status === ResultStatus.READY ||
      (status === ResultStatus.LOADING && !!rows?.length) ||
      (status === ResultStatus.ERROR && !!rows?.length)
    ) {
      return (
        <>
          <CanvasPanel className="explore-chart-panel">
            <div className="dscCanvas__chart">
              <DiscoverChartContainer />
            </div>
          </CanvasPanel>
          <CanvasPanel>
            <ExploreTabs />
          </CanvasPanel>
        </>
      );
    }

    return null;
  };

  const BottomPanel = (
    <EuiResizableContainer
      direction={isMobile ? 'vertical' : 'horizontal'}
      className="explore-layout__bottom-panel"
    >
      {(EuiResizablePanel, EuiResizableButton) => (
        <>
          <EuiResizablePanel
            initialSize={20}
            minSize="260px"
            mode={['collapsible', { position: 'top' }]}
            paddingSize="none"
            style={{ display: showDataSetFields ? 'block' : 'none' }}
          >
            <CanvasPanel testId="dscBottomLeftCanvas">
              <DiscoverPanel />
            </CanvasPanel>
          </EuiResizablePanel>
          <EuiResizableButton style={{ display: showDataSetFields ? 'block' : 'none' }} />
          <EuiResizablePanel
            initialSize={showDataSetFields ? 80 : 100}
            minSize="65%"
            mode="main"
            paddingSize="none"
          >
            <EuiPageBody className="explore-layout__canvas">
              <TopNav {...topNavProps} />
              {renderBottomRightPanel()}
            </EuiPageBody>
          </EuiResizablePanel>
        </>
      )}
    </EuiResizableContainer>
  );

  return (
    <EuiErrorBoundary>
      <div className="mainPage">
        {/* Nav bar structure exactly like data_explorer */}
        <EuiFlexGroup
          direction="row"
          className={showActionsInGroup ? '' : 'mainPage navBar'}
          gutterSize="none"
          alignItems="center"
          justifyContent="spaceBetween"
        >
          {!showActionsInGroup && (
            <EuiFlexItem grow={false}>
              <div ref={topLinkRef} />
            </EuiFlexItem>
          )}
          <EuiFlexItem grow={false}>
            <EuiFlexGroup gutterSize="s" alignItems="center">
              <EuiFlexItem grow={false}>
                <div ref={datasetSelectorRef} />
              </EuiFlexItem>
              <EuiFlexItem grow={false}>
                <div ref={datePickerRef} />
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiFlexItem>
        </EuiFlexGroup>

        <EuiPage className="explore-layout" paddingSize="none" grow={false}>
          <EuiPageBody className="explore-layout__page-body">
            {/* TopNav component - configured like discover */}

            {/* HeaderDatasetSelector component - renders dataset selector in portal */}
            <HeaderDatasetSelector datasetSelectorRef={datasetSelectorRef} />

            <NewExperienceBanner />

            {/* QueryPanel component - only render when IndexPattern is loaded */}
            <div className="dscCanvas__queryPanel">
              {indexPatternLoading ? (
                <div>Loading IndexPattern...</div>
              ) : indexPatternError ? (
                <div>Error loading IndexPattern: {indexPatternError}</div>
              ) : indexPattern ? (
                <QueryPanel
                  datePickerRef={datePickerRef}
                  services={services}
                  indexPattern={indexPattern}
                />
              ) : (
                <div>No IndexPattern available</div>
              )}
            </div>

            {/* Main content area with resizable panels under QueryPanel */}
            {BottomPanel}
          </EuiPageBody>
        </EuiPage>
      </div>
    </EuiErrorBoundary>
  );
};
