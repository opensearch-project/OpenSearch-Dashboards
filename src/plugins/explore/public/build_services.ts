/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { BehaviorSubject } from 'rxjs';
import { CoreStart, PluginInitializerContext } from 'opensearch-dashboards/public';
import { SavedObjectOpenSearchDashboardsServices } from 'src/plugins/saved_objects/public';
import { Storage } from '../../opensearch_dashboards_utils/public';
import { RequestAdapter } from '../../inspector/public';

import { ExploreStartDependencies, ExploreServices } from './types';
import { createSavedExploreLoader } from './saved_explore';
import { getHistory } from './application/legacy/discover/opensearch_dashboards_services';
import { TabRegistryService } from './services/tab_registry/tab_registry_service';
import { VisualizationRegistryService } from './services/visualization_registry_service';

export function buildServices(
  core: CoreStart,
  plugins: ExploreStartDependencies,
  context: PluginInitializerContext,
  tabRegistry: TabRegistryService,
  visualizationRegistry: VisualizationRegistryService,
  isSummaryAgentAvailable$: BehaviorSubject<boolean>
): ExploreServices {
  const services: SavedObjectOpenSearchDashboardsServices = {
    savedObjectsClient: core.savedObjects.client,
    indexPatterns: plugins.data.indexPatterns,
    search: plugins.data.search,
    chrome: core.chrome,
    overlays: core.overlays,
  };
  const savedObjectService = createSavedExploreLoader(services);
  const storage = new Storage(localStorage);

  return {
    addBasePath: core.http.basePath.prepend,
    appName: 'explore',
    capabilities: core.application.capabilities,
    chrome: core.chrome,
    core,
    data: plugins.data,
    docLinks: core.docLinks,
    theme: plugins.charts.theme,
    filterManager: plugins.data.query.filterManager,
    indexPatterns: plugins.data.indexPatterns, // Direct access for convenience
    getSavedExploreById: async (id?: string) => {
      return savedObjectService.get(id);
    },
    getSavedExploreUrlById: async (id: string) => savedObjectService.urlFor(id),
    history: getHistory,
    inspector: plugins.inspector,
    inspectorAdapters: {
      requests: new RequestAdapter(),
    },
    metadata: {
      branch: context.env.packageInfo.branch, // From discover - used for version info
    },
    navigation: plugins.navigation,
    share: plugins.share,
    opensearchDashboardsLegacy: plugins.opensearchDashboardsLegacy,
    urlForwarding: plugins.urlForwarding,
    timefilter: plugins.data.query.timefilter.timefilter,
    toastNotifications: core.notifications.toasts,
    uiSettings: core.uiSettings,
    visualizations: plugins.visualizations,
    storage,
    uiActions: plugins.uiActions,

    // Additional CoreStart properties that are accessed directly
    savedObjects: core.savedObjects,
    notifications: core.notifications,
    http: core.http,
    overlays: core.overlays,

    // From DataExplorerServices (since Explore incorporates DataExplorer functionality)
    store: undefined, // Will be set by the store
    viewRegistry: undefined, // Will be replaced with tabRegistry
    embeddable: plugins.embeddable,
    scopedHistory: undefined, // Will be set by the app
    osdUrlStateStorage: undefined, // Will be set by the app

    // Explore-specific services
    tabRegistry,
    visualizationRegistry,
    expressions: plugins.expressions,

    // For result summary
    isSummaryAgentAvailable$,
    dashboard: plugins.dashboard,
  };
}
