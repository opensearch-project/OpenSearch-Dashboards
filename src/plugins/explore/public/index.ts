/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { PluginInitializerContext } from 'opensearch-dashboards/public';
import './index.scss';

import { ExplorePlugin } from './plugin';

export { ExploreStart } from './application/legacy/discover';
export { SavedExplore, SavedExploreLoader, createSavedExploreLoader } from './saved_explore';

export function plugin(initializerContext: PluginInitializerContext) {
  return new ExplorePlugin(initializerContext);
}
export { ExplorePluginSetup, ExplorePluginStart } from './types';
