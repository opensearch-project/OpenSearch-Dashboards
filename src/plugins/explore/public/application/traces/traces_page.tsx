/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiFlexGroup,
  EuiFlexItem,
  EuiPage,
  EuiPageBody,
  EuiResizableContainer,
  useIsWithinBreakpoints,
} from '@elastic/eui';
import classNames from 'classnames';
import { AppMountParameters } from 'opensearch-dashboards/public';
import React, { memo, useRef } from 'react';
import { IDataPluginServices } from '../../../../data/public';
import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
import { QUERY_ENHANCEMENT_ENABLED_SETTING } from '../../components/constants';
import { Sidebar } from '../../components/sidebar';
import { DISCOVER_LOAD_EVENT, NEW_DISCOVER_LOAD_EVENT, trackUiMetric } from '../../ui_metric';
import DiscoverCanvas from '../legacy/discover/application/view_components/canvas';
import DiscoverContext from '../legacy/discover/application/view_components/context';
import DiscoverPanel from '../legacy/discover/application/view_components/panel';
import './traces_page.scss';

export const TracesPage = ({ params }: { params: AppMountParameters }) => {
  const isMobile = useIsWithinBreakpoints(['xs', 's', 'm']);

  const opensearchDashboards = useOpenSearchDashboards<IDataPluginServices>();
  const { uiSettings } = opensearchDashboards.services;
  const isEnhancementsEnabled = uiSettings?.get(QUERY_ENHANCEMENT_ENABLED_SETTING);
  const showActionsInGroup = uiSettings?.get('home:useNewHomePage');

  const topLinkRef = useRef<HTMLDivElement>(null);
  const datasetSelectorRef = useRef<HTMLDivElement>(null);
  const datePickerRef = useRef<HTMLDivElement>(null);

  trackUiMetric(DISCOVER_LOAD_EVENT);
  if (isEnhancementsEnabled) {
    trackUiMetric(NEW_DISCOVER_LOAD_EVENT);
  }

  const MemoizedPanel = memo(DiscoverPanel);
  const MemoizedCanvas = memo(DiscoverCanvas);

  params.optionalRef = {
    topLinkRef,
    datasetSelectorRef,
    datePickerRef,
  };
  // Render the application DOM.
  return (
    <div className="mainPage">
      <div>WIP: this is traces flavor rendered by the new explore plugin</div>
      {isEnhancementsEnabled && (
        <EuiFlexGroup
          direction="row"
          className={showActionsInGroup ? '' : 'mainPage navBar'}
          gutterSize="none"
          alignItems="center"
          justifyContent="spaceBetween"
        >
          {!showActionsInGroup && (
            <EuiFlexItem grow={false}>
              <div ref={topLinkRef} />
            </EuiFlexItem>
          )}
          <EuiFlexItem grow={false}>
            <div ref={datePickerRef} />
          </EuiFlexItem>
        </EuiFlexGroup>
      )}

      <EuiPage
        className={classNames(
          'deLayout',
          isEnhancementsEnabled && !showActionsInGroup ? 'dsc--next' : undefined
        )}
        paddingSize="none"
        grow={false}
      >
        <DiscoverContext {...params}>
          <EuiResizableContainer direction={isMobile ? 'vertical' : 'horizontal'}>
            {(EuiResizablePanel, EuiResizableButton) => (
              <>
                <EuiResizablePanel
                  initialSize={20}
                  minSize="260px"
                  mode={['collapsible', { position: 'top' }]}
                  paddingSize="none"
                >
                  <Sidebar datasetSelectorRef={datasetSelectorRef}>
                    <MemoizedPanel {...params} />
                  </Sidebar>
                </EuiResizablePanel>
                <EuiResizableButton />

                <EuiResizablePanel initialSize={80} minSize="65%" mode="main" paddingSize="none">
                  <EuiPageBody className="deLayout__canvas">
                    <MemoizedCanvas {...params} />
                  </EuiPageBody>
                </EuiResizablePanel>
              </>
            )}
          </EuiResizableContainer>
        </DiscoverContext>
      </EuiPage>
    </div>
  );
};
