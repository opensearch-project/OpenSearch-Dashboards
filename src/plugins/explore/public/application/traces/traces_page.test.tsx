/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { TracesPage } from './traces_page';
import { AppMountParameters } from 'opensearch-dashboards/public';
import { render } from 'test_utils/testing_lib_helpers';

jest.mock('../legacy/discover/application/view_components/canvas', () => ({
  default: () => <div>canvas</div>,
}));

jest.mock('../legacy/discover/application/view_components/panel', () => ({
  default: () => <div>panel</div>,
}));

jest.mock('../legacy/discover/application/view_components/context', () => ({
  default: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
}));

jest.mock('../../ui_metric', () => ({
  trackUiMetric: jest.fn(),
}));

describe('TracesPage', () => {
  const params: AppMountParameters = {
    element: document.createElement('div'),
    history: {} as any,
    onAppLeave: jest.fn(),
    setHeaderActionMenu: jest.fn(),
    appBasePath: '',
  };

  it('should render the traces page component', () => {
    const { container } = render(<TracesPage params={params} />);

    expect(container).toMatchSnapshot();
  });
});
