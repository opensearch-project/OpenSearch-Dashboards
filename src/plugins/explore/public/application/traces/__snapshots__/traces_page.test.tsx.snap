// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TracesPage should render the traces page component 1`] = `
<div>
  <div
    class="mainPage"
  >
    <div>
      WIP: this is traces flavor rendered by the new explore plugin
    </div>
    <div
      class="euiPage deLayout"
    >
      <div>
        Loading...
      </div>
    </div>
  </div>
</div>
`;
